File: sorting/selection_sort/selection_sort.c
#include<stdio.h>
void swap(int *a, int *b){
    int temp = *a;
    *a = *b;
    *b = temp;
}
void selection_sort(int arr[], int size){
    for(int i=0; i<size; i++){
        int min_index = i;
        for(int j=i+1; j<size; j++){
            if (arr[i]>arr[j]){
                min_index=j;
            }
        }
        swap(&arr[i], &arr[min_index]);
    }
}

int main()
{
    int arr[]={2,1,212,24341,12,9,122};
    int size = sizeof(arr)/sizeof(int);
    selection_sort(arr, size);
    for(int i=0; i<size; i++)
    {
        printf("%d ",arr[i]);   
    }
    return 0;
}

File: sorting/02_selection_sort/selection_sort.py
def selection_sort(ls: list):
    for i in range(len(ls)):
        min_index = i
        for j in range(i+1,len(ls)):
            if ls[min_index]>ls[j]:
                min_index= j
        ls[i],ls[min_index]=ls[min_index],ls[i]
    return ls

ls = [2,1,212,24341,12,9,122]
print(selection_sort(ls))